cmd/frontend/internal/cli/config.go

```diff
@@ -574,48 +601,136 @@ var (
 	symbols.LoadConfig()
 }
 
 func computeSearcherEndpoints() *endpoint.Map {
 	searcherURLsOnce.Do(func() {
-		if len(strings.Fields(searcherURL)) == 0 {
+		addr, err := searcherAddr(os.Environ())
+		if err != nil {
 			searcherURLs = endpoint.Empty(errors.New("a searcher service has not been configured"))
 		} else {
-			searcherURLs = endpoint.New(searcherURL)
+			searcherURLs = endpoint.New(addr)
 		}
 	})
 	return searcherURLs
 }
 
+func searcherAddr(environ []string) (string, error) {
+	const (
+		serviceName = "searcher"
+		port        = "3181"
+	)
+
+	if addr, ok := getEnv(environ, "SEARCHER_URL"); ok {
+		addrs, err := replicaAddrs(deploy.Type(), addr, serviceName, port)
+		return addrs, err
+	}
+
+	// Not set, use the default (service discovery on searcher)
+	return "k8s+http://searcher:3181", nil
+}
+
 func computeIndexedEndpoints() *endpoint.Map {
 	indexedEndpointsOnce.Do(func() {
-		if addr := zoektAddr(os.Environ()); addr != "" {
-			indexedEndpoints = endpoint.New(addr)
+		addr, err := zoektAddr(os.Environ())
+		if err != nil {
+			indexedEndpoints = endpoint.Empty(errors.New("a indexed search service has not been configured"))
 		} else {
-			// It is OK to have no indexed search endpoints.
-			indexedEndpoints = endpoint.Static()
+			if addr != "" {
+				indexedEndpoints = endpoint.New(addr)
+			} else {
+				// It is OK to have no indexed search endpoints.
+				indexedEndpoints = endpoint.Static()
+			}
 		}
 	})
 	return indexedEndpoints
 }
 
-func zoektAddr(environ []string) string {
+func zoektAddr(environ []string) (string, error) {
+	deployType := deploy.Type()
+
+	const port = "6070"
+	var baseName = "indexed-search"
+	if deployType == deploy.DockerCompose {
+		baseName = "zoekt-webserver"
+	}
+
 	if addr, ok := getEnv(environ, "INDEXED_SEARCH_SERVERS"); ok {
-		return addr
+		addrs, err := replicaAddrs(deployType, addr, baseName, port)
+		return addrs, err
 	}
 
 	// Backwards compatibility: We used to call this variable ZOEKT_HOST
 	if addr, ok := getEnv(environ, "ZOEKT_HOST"); ok {
-		return addr
+		return addr, nil
 	}
 
 	// Not set, use the default (service discovery on the indexed-search
 	// statefulset)
-	return "k8s+rpc://indexed-search:6070?kind=sts"
+	return "k8s+rpc://indexed-search:6070?kind=sts", nil
+}
+
+// Generate endpoints based on replica number when set
+func replicaAddrs(deployType, countStr, serviceName, port string) (string, error) {
+	count, err := strconv.Atoi(countStr)
+	// If countStr is not an int, return string without error
+	if err != nil {
+		return countStr, nil
+	}
+
+	fmtStrHead := ""
+	switch serviceName {
+	case "searcher", "symbols":
+		fmtStrHead = "http://"
+	}
+
+	var fmtStrTail string
+	switch deployType {
+	case deploy.Kubernetes, deploy.Helm, deploy.Kustomize:
+		fmtStrTail = fmt.Sprintf(".%s:%s", serviceName, port)
+	case deploy.DockerCompose:
+		fmtStrTail = fmt.Sprintf(":%s", port)
+	default:
+		return "", errors.New("Error: unsupported deployment type: " + deployType)
```

Comment:
nit: errors in go convenetionally start lowercase (to aid readability in wrapping)
```suggestion
		return "", errors.Errorf("unsupported deployment type: %s", deployType)
```